cumulus_library/study_parser.py

```diff
@@ -505,46 +513,49 @@ def run_statistics_builders(
             )
             cursor.execute(insert_query)
 
-    def run_single_table_builder(
+    def run_matching_table_builder(
         self,
         cursor: databases.DatabaseCursor,
         schema: str,
-        name: str,
+        builder: str,
         verbose: bool = False,
         parser: databases.DatabaseParser = None,
     ):
-        """targets a single table builder to run"""
-        if not name.endswith(".py"):
-            name = f"{name}.py"
-        self._load_and_execute_builder(
-            name, cursor, schema, verbose=verbose, drop_table=True, parser=parser
-        )
+        """targets all table builders matching a target string for running"""
+        all_generators = self.get_all_generators()
+        for file in all_generators:
+            if builder and file.find(builder) == -1:
+                continue
```

Comment:
nit: both places you ask for all generators, you immediately filter by `builder` - should that maybe instead be an argument to the get-generators method?